mpegts-segmenter/src/analyzer.rs

```diff
@@ -176,6 +193,56 @@ impl Stream {
                                 0 => 0.0,
                                 num_units_in_tick @ _ => (sps.vui_parameters.time_scale.0 as f64 / (2.0 * num_units_in_tick as f64) * 100.0).round() / 100.0,
                             };
+                            *last_vui_parameters = Some(sps.vui_parameters);
+                        }
+                        h264::NAL_UNIT_TYPE_SUPPLEMENTAL_ENHANCEMENT_INFORMATION => {
+                            let bs = h264::Bitstream::new(nalu.iter().copied());
+                            let mut nalu = h264::NALUnit::decode(bs)?;
+
+                            if let Some(vui_params) = &last_vui_parameters {
+                                let mut rbsp = h264::Bitstream::new(&mut nalu.rbsp_byte);
+                                let sei = h264::SEI::decode(&mut rbsp)?;
+
+                                let acc: Vec<Timecode> = last_timecode.iter().cloned().collect();
+                                let timecodes = sei
+                                    .sei_message
+                                    .iter()
+                                    .flat_map(|sei_message| {
+                                        let mut bs = h264::Bitstream::new(sei_message.payload.iter().copied());
```

Comment:
I think we can move out of sei_message here to simplify things:

```suggestion
                                    .sei_message
                                    .into_iter()
                                    .flat_map(|sei_message| {
                                        let mut bs = h264::Bitstream::new(sei_message.payload);
```